src/Core/Managed/Net40/TaskEx.cs

```diff
@@ -0,0 +1,85 @@
+﻿namespace Microsoft.ApplicationInsights
+{
+    using System;
+    using System.Threading;
+    using System.Threading.Tasks;
+
+    internal static class TaskEx
+    {
+        public static void RethrowIfFailed(this Task task)
+        {
+            if (!task.IsCompleted)
```

Comment:
The method is throw *ifFaied* so if the task is not competed, it has not failed and nothing should happen.